javav2/example_code/dynamodb/src/main/java/com/example/enhanced/dynamodb/BatchWriteItems.java

```diff
@@ -0,0 +1,179 @@
+                    .addAttribute(String.class, a -> a.name("sort")
+                            .getter(Record::getSort)
+                            .setter(Record::setSort)
+                            .tags(primarySortKey()))
+                    .addAttribute(String.class, a -> a.name("attribute")
+                            .getter(Record::getAttribute)
+                            .setter(Record::setAttribute))
+                    .addAttribute(String.class, a -> a.name("attribute2*")
+                            .getter(Record::getAttribute2)
+                            .setter(Record::setAttribute2)
+                            .tags(secondaryPartitionKey("gsi_1")))
+                    .addAttribute(String.class, a -> a.name("attribute3")
+                            .getter(Record::getAttribute3)
+                            .setter(Record::setAttribute3)
+                            .tags(secondarySortKey("gsi_1")))
+                    .build();
+
+    public static void main(String[] args) {
+
+        //Create a DynamoDbClient object
+        Region region = Region.US_EAST_1;
+        DynamoDbClient ddb = DynamoDbClient.builder()
+                .region(region)
+                .build();
+
+        try{
+            // snippet-start:[dynamodb.java2.enhanced.batchitems.main]
+            // Create a DynamoDbEnhancedClient object
+            DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder()
+                    .dynamoDbClient(ddb)
+                    .build();
+
+            //Create a DynamoDbTable object
+            DynamoDbTable<Record> mappedTable = enhancedClient.table("Record", TABLE_SCHEMA);
+
+            //Populate the Table
+            Record record2 = new Record()
+                    .setId("id5")
+                    .setSort("sort-value5")
+                    .setAttribute("val1-5")
+                    .setAttribute2("att2-5")
+                    .setAttribute3("att3-5");
+
+            Record record3 = new Record()
+                    .setId("id6")
+                    .setSort("sort-value6")
+                    .setAttribute("val1-6")
+                    .setAttribute2("att2-6")
+                    .setAttribute3("att3-6");
+
+            BatchWriteItemEnhancedRequest batchWriteItemEnhancedRequest =
+                    BatchWriteItemEnhancedRequest.builder()
+                            .writeBatches(
+                                    WriteBatch.builder(Record.class)
+                                            .mappedTableResource(mappedTable)
+                                            .addPutItem(r -> r.item(record2))
+                                            .build(),
+                                    WriteBatch.builder(Record.class)
+                                            .mappedTableResource(mappedTable)
+                                            .addPutItem(r -> r.item(record3))
+                                            .build())
```

Comment:
These can both be in the same WriteBatch as they are using the same mappedTableResource

Reply:
Nice comment -- i modifed this example to include 

BatchWriteItemEnhancedRequest batchWriteItemEnhancedRequest =
                    BatchWriteItemEnhancedRequest.builder()
                            .writeBatches(
                                    WriteBatch.builder(Record.class)
                                            .mappedTableResource(mappedTable)
                                            **.addPutItem(r -> r.item(record2))
                                            .addPutItem(r -> r.item(record3))**
                                            .build())
                            .build();